src/internal/m365/collection/drive/delta_tree.go

```diff
@@ -437,9 +438,15 @@ func walkTreeAndBuildCollections(
 			With("path_prefix", pathPfx, "path_suffix", parentPath.Elements())
 	}
 
+	customFiles := map[string]*custom.DriveItem{}
+
+	for fID, file := range node.files {
+		customFiles[fID] = custom.ToCustomDriveItem(file)
```

Comment:
Can be done in a future PR, but I wanted to add a note here.

Instead of converting it to custom drive item here, can we do the conversion a little earlier during `enumeratePageOfItems` i.e. before `addFileToTree` and `addFolderToTree` calls?

This way the `custom.DriveItem` can be the source of truth for all drive item properties. Otherwise, if we have a bug in `ToCustomDriveItem`, it'll be hard to discover if we use both `DriveItemable` (during enumeration), and `custom.DriveItem` during item download phase.

It'll also help to track memory usage more accurately, since the usage won't be split between `DriveItemable` and `custom.DriveItem`.

Reply:
sure, don't see why not.

Reply:
@pandeyabs done

Reply:
can do

Reply:
Hmm. We can do that, but we'd still need `DriveItemable` for restore & export operations.